week1/advance/advance.ts

```diff
@@ -0,0 +1,82 @@
+import Dinner from './interface';
+
+const dinner: Dinner = {
+    members: [ 
+        { 
+            name: "권세훈",
+            group: "ob",
+            menu: "봉구스"
+        },
+        {
+            name: "이용택",
+            group: "yb",
+            menu: "봉구스햄"
+        },
+        {
+            name: "장한빛",
+            group: "yb",
+            menu: "봉구스치즈"
+        },
+        {
+            name: "정정빈",
+            group: "ob",
+            menu: "봉구스햄치즈"
+        },
+        {
+            name: "백혜윤",
+            group: "yb",
+            menu: "봉구스제육"
+        },
+        {
+            name:'남지윤',
+            group: "ob",
+            menu: "봉구스치즈제육"
+        },
+        {
+            name:'전희선',
+            group: "yb",
+            menu: "봉구스김치제육"
+        },
+        {
+            name:'양지영',
+            group: "ob",
+            menu: "봉구스닭갈비"
+        },
+        {
+            name:'정준서',
+            group: "yb",
+            menu: "봉구스소불고기"
+        },
+        {
+            name:'조예슬',
+            group: "yb",
+            menu: "봉구스청양불고기"
+        },
+        {
+            name:'한유진',
+            group: "ob",
+            menu: "봉구스불족"
+        },
+        {
+            name:'김민욱',
+            group: "ob",
+            menu: "봉구스맛있새우"
+        }
+    ],
+    shuffle(array) {
+        // fisher-Yates
+        for(let i = 0; i < array.length; i++){
+            let randomIdx = Math.floor(Math.random() * (i+1));
+            let value = array[i];
+            array[i] = array[randomIdx];
+            array[randomIdx] = value;
+        }
+        return array;
+    },
+    organize(array) {
+      this.shuffle(array);
+      console.log(`${array[0].name}, ${array[1].name} 두분은 ${array[0].menu} 나눠드시면됩니다◕‿◕`);
```

Comment:
Javascript에서는 대괄호를 이용한 인덱싱말고도 더 멋지고 괜찮은 인덱싱 방법이 있습니다!

아래와 같이 `Array.at()` 내장 메서드를 사용하시면 됩니다!
```suggestion
      console.log(`${array.at(0).name}, ${array.at(1).name} 두분은 ${array.at(0).menu} 나눠드시면됩니다◕‿◕`);
```


Js에서 대괄호 인덱싱은 Python처럼 `[-1]`을 이용해 제일 뒤 인덱스를 가져올 수 없지만,
`at()` 메서드에서는 가능합니다!

Reply:
.at()을 쓰면 더 깔끔하겠네요. 감사합니다!
저는 맨 마지막 인덱스를 가져올 때, `array[array.length-1]`을 많이 쓰고 있는데, 훈세님은 ` array.at(-1)`로 많이 쓰시나요?
더 가독성이 좋아서 앞으로도 자주 사용하려고 해서 질문 드립니다!

Reply:
@devkwonsehoon 감사합니다.. 알아갑니다 ..